src/components/edit-movie/edit-movie.tsx

```diff
@@ -23,140 +21,194 @@ interface Props {
   action: Actions;
 }
 
+interface Values {
+  title: string;
+  release_date: string;
+  poster_path: string;
+  vote_average: number;
+  genres: string[];
+  runtime: number;
+  overview: string;
+}
+
 export const EditMovie = ({ title, movie, onClose, action }: Props) => {
   const dispatch = useAppDispatch();
 
-  const [movieData, setMovieData] = useState<MovieFormValues | undefined>(
-    movie,
-  );
-
-  const handleSubmit = () => {
-    if (!movieData || !isValidMovie(movieData)) {
+  const onSubmit = (values: Values) => {
+    if (!values || !isValidMovie(values)) {
       return;
     }
 
     if (action === Actions.Add) {
-      dispatch(addMovieAction(movieData));
+      dispatch(addMovieAction(values));
     }
     if (movie?.id) {
-      dispatch(updateMovieAction({ ...movieData, id: movie.id }));
+      dispatch(updateMovieAction({ ...values, id: movie.id }));
     }
     onClose();
   };
   const handleReset = () => {
     onClose();
   };
+  const required = (value: any) => (value ? undefined : 'Required');
 
-  const setData = (
-    e: React.ChangeEvent<HTMLInputElement | HTMLSelectElement>,
-    field: string,
-  ) => {
-    setMovieData((prevState) => ({
-      ...prevState,
-      [field]: e.target.value,
-    }));
+  const initialValues = {
```

Comment:
Since this data is immutable there is no need to re-render it multiple times. Let's wrap with `useMemo`